components/core/JsInterop/modules/components/iconHelper.ts

```diff
@@ -1,11 +1,43 @@
-﻿export class iconHelper {
+﻿import { generate as generateColor } from "@ant-design/colors";
+
+export class iconHelper {
   static createIconFromfontCN(scriptUrl) {
     if (document.querySelector(`[data-namespace="${scriptUrl}"]`)) {
       return;
     }
-    const script = document.createElement('script');
-    script.setAttribute('src', scriptUrl);
-    script.setAttribute('data-namespace', scriptUrl);
+    const script = document.createElement("script");
+    script.setAttribute("src", scriptUrl);
+    script.setAttribute("data-namespace", scriptUrl);
     document.body.appendChild(script);
   }
-}
\ No newline at end of file
+
+  static generateTwotoneSvgIcon(svgImg: string, primaryColor: string): string {
```

Comment:
I think we just need to get the secondary color in js and replace the value in C#.

Reply:
I didn't find solution to add attribute to svg string in C#.  maybe regex or xml?

Reply:
Maybe [Regex.Replace](https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.replace?view=net-6.0&WT.mc_id=DT-MVP-5003987).

Reply:
what about:
```cs
public async Task<string> GetTwotoneSvgIcon(string svgImg, string twotoneColor)
        {
            svgImg = svgImg.Replace("fill=\"#333\"", $"fill=\"primaryColor\"")
            .Replace("fill=\"#E6E6E6\"", $"fill=\"secondaryColor\"")
            .Replace("fill=\"#D9D9D9\"", $"fill=\"secondaryColor\"")
            .Replace("fill=\"#D8D8D8\"", $"fill=\"secondaryColor\"");
            
           // get secondaryColors from js
            var secondaryColors = await _js.InvokeAsync<string>(JSInteropConstants.GenerateColor, twotoneColor);

            var document = XDocument.Load(new StringReader(svgImg));
            var svg = document.Root;
            foreach (var path in svg.Nodes().OfType<XElement>())
            {
                if (path.Attribute("fill")?.Value == "secondaryColor")
                {
                    path.SetAttributeValue("fill", secondaryColors[0]);
                }
                else
                {
                    path.SetAttributeValue("fill", twotoneColor);
                }
            }

            return svg.ToString();
        }
```